server/src/main/java/org/elasticsearch/common/Rounding.java

```diff
@@ -46,58 +48,94 @@
 import java.util.Objects;
 
 /**
- * A strategy for rounding date/time based values.
- *
+ * A strategy for rounding milliseconds since epoch.
+ * <p>
  * There are two implementations for rounding.
- * The first one requires a date time unit and rounds to the supplied date time unit (i.e. quarter of year, day of month)
- * The second one allows you to specify an interval to round to
+ * The first one requires a date time unit and rounds to the supplied date time unit (i.e. quarter of year, day of month).
+ * The second one allows you to specify an interval to round to.
+ * <p>
+ * See <a href="https://davecturner.github.io/2019/04/14/timezone-rounding.html">this</a>
+ * blog for some background reading. Its super interesting and the links are
+ * a comedy gold mine. If you like time zones. Or hate them.
  */
 public abstract class Rounding implements Writeable {
-
     public enum DateTimeUnit {
         WEEK_OF_WEEKYEAR((byte) 1, IsoFields.WEEK_OF_WEEK_BASED_YEAR) {
             long roundFloor(long utcMillis) {
                 return DateUtils.roundWeekOfWeekYear(utcMillis);
             }
+
+            @Override
+            long extraLocalOffsetLookup() {
+                return 604800000L; // 7 days worth of a milliseconds 
```

Comment:
It'd be cute to use `TimeValue.timeValueDays(7).millis()` instead of the comment, and similarly below.

Reply:
:+1: